### ISSUE TITLE
Image augmentation with Sequential doesn't work if there is a Dropout layer.

### ISSUE THREAD

**Issue by dr-mw** [keras-team/keras#17110]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras):
Yes, but very close to F. Chollets newest book "Deep Learning with Python 2nd Edition"
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Reproduced on:
macOS 12.5.1
Ubuntu 22.04.1 LTS
Google Colab
- TensorFlow installed from (source or binary):
Binary on both systems (on Colab I don't know ;) )
- TensorFlow version (use command below):
2.9.2 on both local machines
2.8.2 on Colab
- Python version:
3.9.13 on both local machines
3.7.14 on Colab
- Bazel version (if compiling from source):
N.A.
- GPU model and memory:
MBP M1 Max 64GB
NVDA 3090 24GB

**Describe the problem and behaviour**.
If I perform an image augmentation with the keras Sequential layer, it works only if there was no model with a Dropout layer in the code before.
Here is a reproducable code on Colab. Set the variable in line 12 (reproduce_dropout_error = True), then the Dropout layer will be added and the augmentation doesn't work (all times the same image!). Set it False, and the augmentation works. You need to restart the runtime always!

**Describe the expected behavior**.
The augmentation example should work independently of the existence of the Dropout layer.

**Standalone code to reproduce the issue**.
If you reproduce it on your system, you need to change the image path on line 22 to a directory with some sample images.

https://colab.research.google.com/drive/1HbO8R9c1AJjapzCjB2WM7EIHz2U6H7Zp?usp=sharing

Or here directly:
`
import matplotlib.pyplot as plt
import tensorflow as tf
import sys
print(sys.version)
exit()

from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras.utils import image_dataset_from_directory

reproduce_dropout_error = True

data_augmentation = keras.Sequential([
    layers.RandomFlip("horizontal"),
    layers.RandomRotation(0.1),
    layers.RandomZoom(0.2),
])

train_dataset = image_dataset_from_directory(
    '/',
    image_size=(180, 180),
    batch_size=1)

inputs = keras.Input(shape=(180, 180, 3))
x = data_augmentation(inputs)
x = layers.Rescaling(1./255)(inputs)
x = layers.Conv2D(filters=32, kernel_size=3, activation="relu")(x)
x = layers.MaxPooling2D(pool_size=2)(x)
x = layers.Conv2D(filters=64, kernel_size=3, activation="relu")(x)
x = layers.MaxPooling2D(pool_size=2)(x)
x = layers.Conv2D(filters=128, kernel_size=3, activation="relu")(x)
x = layers.MaxPooling2D(pool_size=2)(x)
x = layers.Conv2D(filters=256, kernel_size=3, activation="relu")(x)
x = layers.MaxPooling2D(pool_size=2)(x)
x = layers.Conv2D(filters=256, kernel_size=3, activation="relu")(x)
x = layers.Flatten()(x)
if reproduce_dropout_error:
  x = layers.Dropout(0.5)(x)
outputs = layers.Dense(1, activation="sigmoid")(x)
model = keras.Model(inputs=inputs, outputs=outputs)

plt.figure(figsize=(10, 10))
for images, _ in train_dataset.take(1):
    for i in range(9):
        augmented_images = data_augmentation(images)
        ax = plt.subplot(3, 3, i + 1)
        plt.imshow(augmented_images[0].numpy().astype("uint8"))
        plt.axis("off")
plt.savefig('test.png')
plt.show()
`

**Comment by dr-mw** [comment]
@tilakrayal ,
I tested your gist. You need to set reproduce_dropout_error = True on line 9, restart the runtime and then the image augmentation will fail (displays 9 times the same image).

**Comment by dr-mw** [comment]
@tilakrayal:
I saw you classified this issue as support. Just to clarify: I don't need any support here. I only want to report an anomalous behavior. To make it easier, I made the example more minimalistic:
https://colab.research.google.com/drive/1HbO8R9c1AJjapzCjB2WM7EIHz2U6H7Zp?usp=sharing

But the core issue remains the same:
1.  Dropout layer in line 9 commented out: augmentation works.
2. Dropout layer in line 9 exists: augmentation fails.
Colab runtime must be restarted, after you (un)comment the layer.

**Comment by tilakrayal** [comment]
@dr-mw,
I haven't found any anomalous behaviour while executing the provided code. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/a223d48ca78ebae7c627012b43a62c6f/untitled688.ipynb). Thank you!

**Comment by tilakrayal** [comment]
@dr-mw,
I tried to execute the code on tf-nightly where the expected behaviour is one original and the other one is augmented image. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/809fa9efbb0155b9596f4d67ee66c548/untitled696.ipynb). Thank you!

**Comment by dr-mw** [comment]
@tilakrayal,
Did you fix this bug in the nightly build, or was it already fixed?
Thank you.
